### DISCUSSION TITLE
Function-npm node not working

### DISCUSSION THREAD

**Discussion by githubniwis** [ntsaini/node-red-contrib-function-npm#34] [Q&A]
Mac mini M1 Big Sur 11.2.3
Hi,
After deploying a function-npm node, the terminal gives the following message:

"(node:21925) UnhandledPromiseRejectionWarning: TypeError: must call as: npm.load(callback)
    at load (/usr/local/lib/node_modules/npm/lib/npm.js:123:13)
    at internal/util.js:297:30
    at new Promise (<anonymous>)
    at load (internal/util.js:296:12)
    at process.<anonymous> (/Volumes/T5/Users/.node-red/node_modules/npm-install-to/lib/npm-install-child.js:7:11)
    at process.emit (events.js:315:20)
    at emit (internal/child_process.js:903:12)
    at processTicksAndRejections (internal/process/task_queues.js:81:21)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:21925) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:21925) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code."

I used the upper/lowercase example from GitHub. Under the node appears a blue icon with the text "installing".
This is not an urgent problem, since I was just experimenting, but this node has always been one of my favourites in prior OS's.
Any help?

**Comment by ntsaini** [comment]
Can you please tell me version of node-red and the node-red-contrib-function-npm you are using? I don't have a Mac OS machine to test but if you can clone the source on your machine and run the unit tests locally 'npm test', it would be helpful to know the outcome.

**Reply by githubniwis** [reply]
The latest for both.
After some searching, I found that to use node packages you can also use  the node-red-contrib-function and functionGlobalContext{} in the settings.js file, which I didn't know. So my problem is solved.
Thanks for your time!
